### ISSUE TITLE
Double indentation on C++ initialization arguments

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1268]
```cpp
        floatNx3 randomRotationMatrices[3];
        if (rotationRandomnessX > epsilon() || rotationRandomnessY > epsilon())
        {
            floatNx3 rotationEuler = floatNx3(
                    (GenerateRandom(randomSeed + intN(kParticleSystemExternalForcesRotationRandomnessXId)) * 2 - 1) * rotationRandomnessX,
                    (GenerateRandom(randomSeed + intN(kParticleSystemExternalForcesRotationRandomnessYId)) * 2 - 1) * rotationRandomnessY,
                    floatN(ZERO));
            eulerToMatrix(rotationEuler, randomRotationMatrices);

            toForce = mul(randomRotationMatrices, toForce);
        }
```

Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60022-UNI-18439.cpp" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60022-UNI-18439.cpp"
index 8d467c6..446501d 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60022-UNI-18439.cpp"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60022-UNI-18439.cpp"
@@ -1,12 +1,12 @@
 floatNx3 randomRotationMatrices[3];
 if (rotationRandomnessX > epsilon() || rotationRandomnessY > epsilon())
 {
 // Parameters are being double indented.
     floatNx3 rotationEuler = floatNx3(
-        (GenerateRandom(randomSeed + intN(kParticleSystemExternalForcesRotationRandomnessXId)) * 2 - 1) * rotationRandomnessX,
-        (GenerateRandom(randomSeed + intN(kParticleSystemExternalForcesRotationRandomnessYId)) * 2 - 1) * rotationRandomnessY,
-        floatN(ZERO));
+            (GenerateRandom(randomSeed + intN(kParticleSystemExternalForcesRotationRandomnessXId)) * 2 - 1) * rotationRandomnessX,
+            (GenerateRandom(randomSeed + intN(kParticleSystemExternalForcesRotationRandomnessYId)) * 2 - 1) * rotationRandomnessY,
+            floatN(ZERO));
     eulerToMatrix(rotationEuler, randomRotationMatrices);
 
     toForce = mul(randomRotationMatrices, toForce);
 }
```

Debug support files:
---
[60022-UNI-18439.cpp](https://gist.githubusercontent.com/mihaipopescu/64132362b93233ae55b623e424851959/raw/8d467c67af6f5b4ff3ba167f2089fa0c436d8c03/60022-UNI-18439.cpp)
[60022-UNI-18439.cpp.log](https://gist.githubusercontent.com/mihaipopescu/64132362b93233ae55b623e424851959/raw/67abd1d98722929ba1a69b6d3158b5f1f759ae9e/60022-UNI-18439.cpp.log)
[60022-UNI-18439.cpp.unc](https://gist.githubusercontent.com/mihaipopescu/64132362b93233ae55b623e424851959/raw/653dedc8eb3c0e9d445490fe48144c0e865fe640/60022-UNI-18439.cpp.unc)
[Uncrustify.Cpp.cfg](https://gist.githubusercontent.com/mihaipopescu/64132362b93233ae55b623e424851959/raw/23d62d9dba5caf352fdc1793792e98eed4b81929/Uncrustify.Cpp.cfg)
[UNI-18439.cpp](https://gist.githubusercontent.com/mihaipopescu/64132362b93233ae55b623e424851959/raw/0c40e831b9ded27094f2657a073c139fbc3acec6/UNI-18439.cpp)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by CDanU** [comment]
I haven't gone through all issues yet but it seems a bunch of them is related to indentions.
While tying to solve #1218 I made this branch: [code_width](https://github.com/CDanU/uncrustify/commits/fix-code_width). Contrary to the name the goal was to fix the width for func call params. The solution works somewhat better but fights with the rest of the code inside indent.cpp (mainly everything that gets done by [uncrustify.cpp - indent_text() - L1890-L1907](https://github.com/uncrustify/uncrustify/blob/9e66347a1247923c139c8a97310affb31d049966/src/uncrustify.cpp#L1890-L1907))

**Comment by mihaipopescu** [comment]
Yes, a quite a few are double indentation related but they are different. We're not using `code_width` at all. You think it's the same code that deals with width of func call params ? What is the option for that ?

**Comment by CDanU** [comment]
Ohh ok, I saw the function call and thought of the the code I wrote. No the changes I had made are only triggered when `code_width` is violated. But the problems here most likely are still caused by indent_text().

**Comment by guy-maurel** [comment]
Please push Uncrustify.Common-Cpp.cfg

**Comment by guy-maurel** [comment]
Should it be controlled by the option `use_indent_continue_only_once = true` ?

**Comment by CDanU** [comment]
We have both configs in the repo: 
[Uncrustify.Common-CStyle.cfg](https://github.com/uncrustify/uncrustify/blob/master/tests/config/Uncrustify.Common-CStyle.cfg)

[Uncrustify.Common-Cpp.cfg](https://github.com/uncrustify/uncrustify/blob/master/tests/config/Uncrustify.Common-Cpp.cfg)

**Comment by mihaipopescu** [comment]
Yeah, I noticed that they're missing and I started to reduce the configs but didn't finished that so I pushed it like this. But, as mentioned in the main discussion thread #1223 the config can be extracted from the dump `.unc` file.

**Comment by CDanU** [comment]
Caused by:
```sql
# Align continued statements at the '='. Default=True
# If False or the '=' is followed by a newline, the next line is indent one tab.
indent_align_assign             = false    # false/true
```

**Comment by mihaipopescu** [comment]
Cool! Are you finding these with the option reducer ?

**Comment by CDanU** [comment]
No, I just paste the config in my webtool, manually reduce the config in halve and check the output until I find the problem.

**Comment by gmaurel** [comment]
Fixed at 2c412d8929c2c11e50660ad82e95271831e4a98e, Dec 17 18:33:16 2017
